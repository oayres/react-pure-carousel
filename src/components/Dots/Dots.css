.list {
  position: relative;
  margin: 0;
  top: -10px;
  padding: 0;
}

.item {
  list-style: none;
  display: inline-block;
}

.button {
  border: 0;
  background: transparent;
  color: black;
  cursor: pointer;
  padding: 10px;
  outline: 0;
  font-size: 24px;
  opacity: 0.5;
}

.active {
  opacity: 1;
}
