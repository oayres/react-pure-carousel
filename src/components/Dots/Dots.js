import React from 'react'
const PropTypes = require('prop-types')
import styles from './Dots.css'

class Dots extends React.Component {
  getIndexes (count, inc) {
    const arr = []

    for (let i = 0; i < count; i += inc) {
      arr.push(i)
    }

    return arr
  }

  renderDots (items, currentSlide, clickHandler) {
    const dots = []

    items.map(index => {
      const classes = [styles.button]

      if (currentSlide === index) {
        classes.push(styles.active)
      }

      dots.push(
        <li className={styles.item} key={index}>
          <button className={classes.join(' ')} onClick={clickHandler.bind(null, index)}>
            &bull;
          </button>
        </li>
      )
    })

    return dots
  }

  render () {
    const indexes = this.getIndexes(this.props.slideCount, this.props.slidesToScroll)

    return (
      <ul className={styles.list}>
        { this.renderDots(indexes, this.props.currentSlide, this.props.goToSlide) }
      </ul>
    )
  }
}

Dots.PropTypes = {
  slideCount: PropTypes.number,
  slidesToScroll: PropTypes.number,
  currentSlide: PropTypes.number,
  goToSlide: PropTypes.func.isRequired
}

Dots.defaultProps = {
  slideCount: 1,
  slidesToScroll: 1
}

export default Dots
