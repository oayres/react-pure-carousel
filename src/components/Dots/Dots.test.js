import Dots from './Dots'

describe('The Dots component', () => {
  const minProps = {
    goToSlide: jest.fn()
  }

  it('renders without exploding', () => {
    const wrapper = shallow(<Dots {...minProps} />)
    expect(wrapper.length).toEqual(1)
    expect(snapshot(wrapper)).toMatchSnapshot()
  })
})
