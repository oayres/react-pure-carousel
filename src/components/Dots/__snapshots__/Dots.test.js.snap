exports[`The Dots component renders without exploding 1`] = `
<ul
  className="list">
  <li
    className="item">
    <button
      className="button"
      onClick={[Function]}>
      â€¢
    </button>
  </li>
</ul>
`;
